Source/Modules/r.cxx

```diff
@@ -2087,9 +2087,21 @@ int R::functionWrapper(Node *n) {
       Node * parent = Getattr(n, "parentNode");
       String * smartname = Getattr(parent, "feature:smartptr");
       if (smartname) {
-	smartname = getRClassName(smartname, 1, 1);
-	Replaceall(tm, "$R_class", smartname);
-	Delete(smartname);
+        String * classtypeobj = NewString(Getattr(parent, "classtypeobj")); // this is the correct name, e.g. ClassName(int)
```

Comment:
`classtypeobj` isn't modified below so making a copy here is unnecessary.

Reply:
ok, should be fixed in the commit "fixes from code review"